### PR TITLE
remove ServiceClassHolder and use the ApplicationModel replacedly

### PR DESCRIPTION
**Pull request by Jeff-Lv**
## What is the purpose of the change

remove ServiceClassHolder which was created for only REST and use the ApplicationModel replacedly

## Verifying this change

XXXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

- [x] Make sure there is a [GITHUB_issue](https://github.com/apache/incubator-dubbo/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a GITHUB issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Format the pull request title like `[Dubbo-XXX] Fix UnknownException when host config not exist #XXX`. Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/incubator-dubbo/tree/master/dubbo-test).
- [ ] Run `mvn clean install -DskipTests` & `mvn clean test-compile failsafe:integration-test` to make sure unit-test and integration-test pass.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/incubator-dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Comment by carryxyh** [comment]
deal the conflicts first.

File: dubbo-demo/dubbo-demo-consumer/src/main/resources/META-INF/spring/dubbo-demo-consumer.xml [hunk 1]
```diff
@@ -30,6 +30,8 @@
 
     <!-- generate proxy for the remote service, then demoService can be used in the same way as the
     local regular interface -->
-    <dubbo:reference id="demoService" check="false" interface="org.apache.dubbo.demo.DemoService"/>
+    <dubbo:reference id="demoService" check="false" interface="org.apache.dubbo.demo.DemoService">
+        <dubbo:method name="findFoo" async="true" />
```
[the remark this answers is not in the document]
Comment (Jeff-Lv) [reply dubbo-demo/dubbo-demo-consumer/src/main/resources/META-INF/spring/dubbo-demo-consumer.xml]: > What is this config doing?

test the async config of dubbo.

[the remark this answers is not in the document]
Comment (carryxyh) [reply dubbo-demo/dubbo-demo-consumer/src/main/resources/META-INF/spring/dubbo-demo-consumer.xml]: I konw. But this is not test module.
The demo only maintains the simplest example. If you need to test, please test it in your local, don't submit it to the demo.

**Comment by carryxyh** [comment]
Hi
I have already approved this pr, but I think this pr needs to be reviewed by more people.
@beiwei30 @chickenlj 
Could u pls have a look?

**Pull request merged**